src/Laratrust/Traits/LaratrustRoleTrait.php

```diff
@@ -226,4 +210,21 @@ public function flushCache()
     {
         Cache::forget('laratrust_permissions_for_role_' . $this->getKey());
     }
+
+    /**
+     * @param $permission
+     * @return mixed
+     */
+    private function getIdFor($permission)
+    {
+        if (is_object($permission)) {
+            $permission = $permission->getKey();
+        }
+
+        if (is_array($permission)) {
+            $permission = $permission['id'];
+        }
```

Comment:
better "elseif() {} else {throw exception}"?

so "$permission" is always set for use as return value.